docs/source/gateway.rst

```diff
@@ -0,0 +1,763 @@
+
+The `MLflow Gateway` service relies on a user-provided configuration file, written in YAML,
+that defines the routes and providers available to the service. The configuration file dictates
+how the gateway interacts with various language model providers and determines the end-points that
+users can access.
+
+The configuration file includes a series of sections, each representing a unique route.
+Each route section has a name, a type, and a model specification, which includes the model
+provider, name, and configuration details. The configuration section typically contains the base
+URL for the API and an environment variable for the API key.
+
+Here is an example of a simple single-route configuration:
+
+.. code-block:: yaml
+
+    routes:
+      - name: chat
+        type: llm/v1/chat
+        model:
+          provider: openai
+          name: gpt-3.5-turbo
+          config:
+            openai_api_base: https://api.openai.com/v1
+            openai_api_key: $OPENAI_API_KEY
+
+
+In this example, we define a route named ``chat`` that corresponds to the ``llm/v1/chat`` type, which
+will use the ``gpt-3.5-turbo`` model from `OpenAI` to return query responses from the `OpenAI` service.
+
+The beauty of the Gateway Configuration lies in its flexibility and ease of use.
+If the configuration file is updated at the target location, the `Mlflow Gateway` service can hot-swap
+routes, allowing new providers or model types to be incorporated without requiring changes to
+the applications interfacing with the gateway. The hot-swap functionality is autonomous and
+requires no server downtime or manual restarting of the service.
+This allows for simple, seamless changes to the underlying models and providers while keeping
+your applications steady and reliable.
+
+In order to define an API KEY for a given provider, there are two primary options:
+
+1. Directly include it in the YAML configuration file.
+2. Use an environment variable to store the API key and reference it in the YAML configuration file.
+
+If you choose to include the API key directly, replace ``$OPENAI_API_KEY`` in the YAML file with your
+actual API key.
+
+If you prefer to use an environment variable (recommended), you can define it in your shell
+environment. For a Linux system, you can set the environment variable in the ``.bashrc`` file:
+
+1. Open your ``.bashrc`` file in a text editor. For example, using nano:
+
+.. code-block:: bash
+
+     nano ~/.bashrc
+
+2. Scroll down to the end of the file and add the following line:
+
+.. code-block:: bash
+
+     export OPENAI_API_KEY="your_openai_api_key"
+
+Replace "your_openai_api_key" with your actual `OpenAI` API key.
+
+3. Save the file and exit the text editor. If you're using nano, you can do this by pressing
+Ctrl+X, then Y to confirm that you want to save the changes, and finally Enter to confirm
+the file name.
+
+4. Reload the .bashrc file with the following command:
+
+.. code-block:: bash
+
+     source ~/.bashrc
```

Comment:
I don't think we need to mention `nano` or `.bashrc`. This can make users wonder "Does the key need to set in .bashrc?" or "Does the gateway only support bash"?

Reply:
We can just say how to set an API key.

Reply:
great point. Removing.